openupgradelib/openupgrade.py

```diff
@@ -917,6 +918,104 @@ def rename_models(cr, model_spec):
+    """
+    Update model references for models that have merged to an existing model.
+    :param old_model: old model
+    :param new_model: destination model
+    :param ref_field: name of the field in new model that references old model.
+
+    WARNING: This method doesn't move the records from old_model to new_model
+    tables. You should have to that previously in the migration scripts.
+    """
+    logger.info("model %s: merging to %s", old_model, new_model)
+    table = new_model.replace('.', '_')
+    renames = [
+        ('ir_attachment', 'res_model', 'res_id', ''),
+        ('ir_model_data', 'model', 'res_id', ''),
+        ('ir_filters', 'model_id', '', ''),
+        ('ir_exports', 'resource', '', ''),
+    ]
+    if is_module_installed(cr, 'mail'):
+        renames += [
+            ('mail_message', 'model', 'res_id', ''),
+            ('mail_message_subtype', 'res_model', '', ''),
+            ('mail_activity', 'res_model', 'res_id', 'res_model_id'),
+            ('mail_activity_type', '', '', 'res_model_id'),
+            ('mail_template', 'model', '', 'model_id'),
+            ('mail_alias', '', '', 'alias_model_id'),
+            ('mail_alias', '', 'alias_parent_thread_id',
+             'alias_parent_model_id'),
+            # mail_followers: special case handled below
+        ]
+    for rename in renames:
+        if not table_exists(cr, rename[0]):
+            continue
+        old_model_id, new_model_id = "", ""
+        query_2a, query_2b, query_3 = "", "", ""
+        query_1a = """{field1} = %(new_value1)s"""
+        query_1b = """t.{field1} = %(old_value1)s"""
+        if rename[2]:
+            query_2a = """, {field2} = mt.id
+                FROM {model_table} mt"""
+            query_2b = """ AND mt.{field} = t.{field2}"""
+        if rename[3]:
+            pre_query = """
+                SELECT id
+                FROM ir_model
+                WHERE model = %(value1)s"""
+            cr.execute(sql.SQL(pre_query), {
+                "value1": new_model,
+            })
+            new_model_id = cr.fetchone()[0]
+            query_3 = """, {field3} = %(new_value2)s"""
+            if not rename[1]:
+                cr.execute(sql.SQL(pre_query), {
+                    "value1": old_model,
+                })
+                old_model_id = cr.fetchone()[0]
+                query_1a = ""
+                query_1b = """t.{field3} = %(old_value2)s"""
+                query_3 = """{field3} = %(new_value2)s"""
+        query = """
+            UPDATE {table} t
+            SET """ + query_1a + query_3 + query_2a + """
+            WHERE """ + query_1b + query_2b
+        logged_query(cr, sql.SQL(query).format(
+            model_table=sql.Identifier(table),
+            table=sql.Identifier(rename[0]),
+            field1=sql.Identifier(rename[1]),
+            field2=sql.Identifier(rename[2]),
+            field3=sql.Identifier(rename[3]),
+            field=sql.Identifier(ref_field)
```

Comment:
Here too, the query would be more clear to me I think if the `field1`, `field2`(etc) keys were renamed to something like `model_name_column` etc. What do you think?